ByteCoin/ByteCoin/Model/CoinManager.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  CoinManager.swift
+//  ByteCoin
+//
+//  Created by Angela Yu on 11/09/2019.
+//  Copyright © 2019 The App Brewery. All rights reserved.
+//
+
+import Foundation
+
+protocol CoinManagerDelegate {
+    func didUpdatePrice(price: String, currency: String)
+    func didFailWithError(error: Error)
+}
+
+struct CoinManager {
+    var delegate: CoinManagerDelegate?
+    let baseURL = "https://rest.coinapi.io/v1/exchangerate/BTC"
+    let apiKey = "0AF7E62A-40B6-49DA-985B-99E5D77B55A3"
+    
+    let currencyArray = ["AUD", "BRL","CAD","CNY","EUR","GBP","HKD","IDR","ILS","INR","JPY","MXN","NOK","NZD","PLN","RON","RUB","SEK","SGD","USD","ZAR"]
+
+    func getCoinPrice(for currency: String) {
```

Comment:
サンプルコードの問題ではありますが、swiftではguardによる早期リターンでネストのレベルを浅くする書き方が好まれます。

```suggestion
    func getCoinPrice(for currency: String) {
        let urlString = "\(baseURL)/\(currency)?apikey=\(apiKey)"
        
        guard let url = URL(string: urlString) else { return }

        let session = URLSession(configuration: .default)

        let task = session.dataTask(with: url) { (data, response, error) in
            if let error = error {
                self.delegate?.didFailWithError(error: error)
                return
            }
                
            guard let safeData = data, let price = parseJSON(safeData) else { return }
            let priceString = String(format: "%.2f", price)
            self.delegate?.didUpdatePrice(price: priceString, currency: currency)
        }
        task.resume()
    }
```
```